src/net/sourceforge/kolmafia/textui/Parser.java

```diff
@@ -2967,18 +2966,33 @@ private Evaluable parseExpression(final BasicScope scope, final Operator previou
         oper = new Operator(this.makeLocation(operator), operator.content, this);
         this.readToken(); // -
         if ((lhs = this.parseEvaluable(scope)) == null) {
-          throw this.parseException("Value expected");
+          Location errorLocation = this.makeLocation(this.currentToken());
+
+          throw this.parseException(errorLocation, "Value expected");
         }
 
         lhs = new Operation(lhs, oper);
+        if (!lhs.getType().equals(DataTypes.INT_TYPE)
+            && !lhs.getType().equals(DataTypes.FLOAT_TYPE)) {
+          throw this.parseException(
+              lhs.getLocation(), "\"-\" operator requires an integer or float value");
+        }
       }
-    } else if (operator.equals("remove")) {
+    } else if (operator.equalsIgnoreCase("remove")) {
+      // Due to the workings of Operator, we need 'remove' to be case-sensitive
+      if (!operator.equals("remove")) {
+        throw this.parseException(operator, "Bad 'remove' statement");
```

Comment:
This will only ever happen if `remove` is not lowercase. We could say as much in the exception text.

Reply:
Though could we not just overwrite `operator` with lowercase "remove" here?

Reply:
we could, and we can!